qa/rpc-tests/spentindex.py

```diff
@@ -0,0 +1,207 @@
+        connect_nodes(self.nodes[0], 1)
+        connect_nodes(self.nodes[0], 2)
+
+        self.is_network_split = False
+        self.sync_all()
+
+    def run_test(self):
+        self.nodes[0].generate(105)
+        self.sync_all()
+
+        chain_height = self.nodes[1].getblockcount()
+        assert_equal(chain_height, 105)
+
+        # Test getrawtransaction changes and the getspentinfo RPC
+
+        # send coinbase to address addr1
+        addr1 = self.nodes[1].getnewaddress()
+        txid1 = self.nodes[0].sendtoaddress(addr1, 2)
+        self.sync_all()
+        block_hash1 = self.nodes[0].generate(1)
+        self.sync_all()
+
+        # send from addr1 to addr2
+        # (the only utxo on node 1 is from address addr1)
+        addr2 = self.nodes[2].getnewaddress()
+        txid2 = self.nodes[1].sendtoaddress(addr2, 1)
+        self.sync_all()
+
+        # addr1 to addr2 transaction is not confirmed, so it has no height
+        tx2 = self.nodes[2].getrawtransaction(txid2, 1)
+        assert('height' not in tx2)
+
+        # confirm addr1 to addr2 transaction
+        block_hash2 = self.nodes[0].generate(1)
+        self.sync_all()
+
+        # Restart all nodes to ensure index files are saved to disk and recovered
+        stop_nodes(self.nodes)
+        wait_bitcoinds()
+        self.setup_network()
+
+        # Check new fields added to getrawtransaction
+        tx1 = self.nodes[2].getrawtransaction(txid1, 1)
+        assert_equal(tx1['vin'][0]['value'], 10) # coinbase
+        assert_equal(tx1['vin'][0]['valueSat'], 10*COIN)
+        # we want the non-change (payment) output
+        vout = filter(lambda o: o['value'] == 2, tx1['vout'])
+        n = vout[0]['n']
+        assert_equal(vout[0]['spentTxId'], txid2)
+        assert_equal(vout[0]['spentIndex'], 0)
+        assert_equal(vout[0]['spentHeight'], 107)
+        assert_equal(tx1['height'], 106)
+
+        tx2 = self.nodes[2].getrawtransaction(txid2, 1)
+        assert_equal(tx2['vin'][0]['address'], addr1)
+        assert_equal(tx2['vin'][0]['value'], 2)
+        assert_equal(tx2['vin'][0]['valueSat'], 2*COIN)
+        # since this transaction's outputs haven't yet been
+        # spent, these fields should not be present
+        assert('spentTxId' not in tx2['vout'][0])
+        assert('spentIndex' not in tx2['vout'][0])
+        assert('spentHeight' not in tx2['vout'][0])
+        assert_equal(tx2['height'], 107)
+
+        # Given a transaction output, getspentinfo() returns a reference
+        # to the (later, confirmed) transaction that spent that output,
+        # that is, the transaction that used this output as an input.
+        spentinfo = self.nodes[2].getspentinfo({'txid': txid1, 'index': n})
+        assert_equal(spentinfo['height'], 107)
+        assert_equal(spentinfo['index'], 0)
+        assert_equal(spentinfo['txid'], txid2)
+
+        # Some invalid argument checks...
```

Comment:
I don't know if I need all these checks here ... I wrote them before I was aware of `src/test/rpc_tests.cpp` (you can run the tests I added there as follows: `src/test/test_bitcoin --run_test=rpc_tests/rpc_insightexplorer`). Are these okay, or should I remove them?